precompiles/ics20/types.go

```diff
@@ -143,22 +143,32 @@ func NewMsgTransfer(method *abi.Method, args []interface{}) (*transfertypes.MsgT
 		Amount: sdk.NewIntFromBigInt(amount),
 	}
 
-	msg := &transfertypes.MsgTransfer{
-		SourcePort:       sourcePort,
-		SourceChannel:    sourceChannel,
-		Token:            token,
-		Sender:           sdk.AccAddress(sender.Bytes()).String(), // convert to bech32 format
-		Receiver:         receiver,
-		TimeoutHeight:    input.TimeoutHeight,
-		TimeoutTimestamp: timeoutTimestamp,
-		Memo:             memo,
+	msg, err := NewMsgTransfer(sourcePort, sourceChannel, token, sdk.AccAddress(sender.Bytes()).String(), receiver, input.TimeoutHeight, timeoutTimestamp, memo)
+	if err != nil {
+		return nil, common.Address{}, err
 	}
 
+	return msg, sender, nil
+}
+
+// NewMsgTransfer creates a new MsgTransfer message
+func NewMsgTransfer(sourcePort, sourceChannel string, coin sdk.Coin, senderAddress, receiverAddress string, timeoutHeight clienttypes.Height, timeoutTimestamp uint64, memo string) (*transfertypes.MsgTransfer, error) {
```

Comment:
I would like to avoid passing so many parameters as that creates a maintainability burden. But not sure if theres a way around this here that makes sense. The NewMsgTransfer function seems innecseary since it just creates the struct but I guess thats fine for now

Reply:
Thanks for the review @facs95 What standard we should use to split the signature? Isn't something that the linter should care about?

Reply:
I agree on both the name and the too many params. However, we need to build this message for all precompiles that use ics20. An alternative is to build directly the type and validate without relying on this function